packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -40,6 +40,45 @@ export interface ISpinButtonState {
    * as active when up/down arrow is pressed
    */
   keyboardSpinDirection: KeyboardSpinDirection;
+
+  /**
+   * The calculated precision for the value.
+   */
+  precision: number;
+}
+
+/**
+ *
+ * @param value
+ */
+export function calculatePrecision(value: number | string): number {
+  /**
+   * Group 1:
+   * [1-9]([0]+$) matches trailing zeros
+   * Group 2:
+   * \.([0-9]*) matches all digits after a decimal point.
+   */
+  const groups = /[1-9]([0]+$)|\.([0-9]*)/.exec(String(value));
+  if (!groups) {
+    return 0;
+  }
+  if (groups[1]) {
+    return -groups[1].length;
+  }
+  if (groups[2]) {
+    return groups[2].length;
+  }
+  return 0;
+}
+
+/**
+ *
+ * @param value The value that is being rounded.
+ * @param precision The number of decimal places to round the number to
+ */
+export function precisionRound(value: number, precision: number, base: number = 10): number {
+  const exp = Math.pow(base, precision);
+  return Math.round(value * exp) / exp;
```

Comment:
Why not just do `return value.toFixed(precision));` toFixed will round decimals for you and even returns as a string.

Reply:
toFixed doesn't support negative precision.